src/mapper/github.mapping.ts

```diff
@@ -0,0 +1,58 @@
+import { getReposList, getRepoInfo } from '../client-services/github.client-service'
+
+interface RepoInfo {
+    name: string;
+    description?: string;
+    url: string;
+    languages: string[];
+    language: string;
+    contributors: { 
+        name: string; 
+        avatarUrl: string 
+    }[];
+}
+
+export async function map(): Promise<RepoInfo[]> {
```

Comment:
poor function naming and lack of reusuablilty. 
e.g. how are you going to fetch information of a subset of repos and not the whole set (ie. for paginated results etc)? this function is written in such a way that is very hard to test and tightly coupled.

Reply:
@umsu2 
Thanks, Yang for your feedback. The first function name was githubMapping and I changed it due to another provided feedback. If you think it is nonsense, I can revert back. 
The problem with the current version of the code is that I used for loop instead of mep which makes the function a bit complex. I try to fix that and push the changes. If you have any further concerns about the code block, please let me know.

Reply:
@SamHuo213  I am not sure what is the quality we want for these PRs. If we are just aiming to get things that `works`. then this is fine. but if we want to have good maintainable code and devs want to learn more about how to write clean code then this would be one of those candidate. let me know where we want to put the emphasis on.

Reply:
@PooriaT , @umsu2 ,

I think the previous name is not a good name. I had suggested map but open to other.

Keep in mind that there is enough context here, it takes a type and returns a specific type.

However, I do see a potential issue here with this convention. If a client uses different maps then they might be force to alias the method name.

Some suggestions:
1) map[DestinationObject] e.g. mapProject(gitHubRepo)
2) use a static class as a name space
3) use templates

I am a fan of C# automapper, so this follows that pattern in a way: https://github.com/AutoMapper/AutoMapper

Reply:
@umsu2 ,

@PooriaT is going to refactor this method. He will take input objects (github objects) and transform it into a list of projects or project.

e.g.
`
mapProject(githubObject) {
    return {
        ...
    };
}

mapProjects(githubObjects) {
    returns githubObjects.map(x => mapProject(x));
}
`

Reply:
@umsu2 , I think there should be a balance.

However, in this certain area we can focus on having clean code.